scripts/keyframes.py

```diff
@@ -0,0 +1,422 @@
+                break
+        # end for each frame range
+    # end for each footprint
+
+    return keyframes
+
+def getConfigPath(filename):
+    """Look for a named configuration file in the user's home directory, or
+    in the current working directory, or in any parent of the current working directory.
+    Returns the filename of the first configuration file found, or None if none existed."""
+    for directory in [pathlib.Path.home(), pathlib.Path.cwd()] + list(pathlib.Path.cwd().parents):
+        config_path = directory / filename
+        if config_path.is_file():
+            return config_path
+
+    return None
+
+def makeStubReferenceImages(directory, keyframes, filename_format='screenshot_frame_{keyframe}.bmp'):
+    """Given a directory and a set of keyframes, create that directory
+    (if needed) and create a file in that directory for every keyframe,
+    using the filename_format string provided."""
+    image_directory = pathlib.Path(directory)
+    if not image_directory.exists():
+        image_directory.mkdir(parents=True)
+    elif not image_directory.is_dir():
+        log.warning("stub reference images directory exists but is not a directory: %s",
+                    directory)
+        return
+
+    for keyframe in keyframes:
+        image_filename = filename_format.format(keyframe=keyframe)
+        stub_image_filename = image_directory/image_filename
+        if stub_image_filename.exists():
+            log.warning('not creating stub image %s, as the file already exists', stub_image_filename)
+        else:
+            log.info('creating stub image file %s', stub_image_filename)
+            stub_image_filename.write_text('')
+
+##################################################################
+# Main program.
+def main():
+    """Main program function."""
+    LOG_LEVELS = ['CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG']
+    LOG_FORMAT = '%(levelname)s: %(message)s'
+
+    parser = argparse.ArgumentParser(description='identify interesting keyframes')
+    parser.add_argument('--log-level', type=str.upper, default='INFO', choices=LOG_LEVELS,
+                        help='the desired logging level [%(default)s]')
+    common = parser.add_argument_group('common arguments')
+    common.add_argument('-n', type=int, default=10,
+                        help="how many keyframes to return [%(default)s]")
+
+    filter_descriptions = [f'{x.name} ({x.description})' for x in Filter.__subclasses__()]
+    filter_help = ('comma-separated list of keyframe filters to use [none]; available filters are: '
+                   + '; '.join(filter_descriptions))
+    common.add_argument('-f', '--filters', help=filter_help)
+
+    common.add_argument('--stub-reference-images', metavar='DIRECTORY',
+                        help='if set, names a directory where stub reference images will be created')
+
+    common.add_argument('--save-command-counts', metavar='JSONFILE',
```

Comment:
Thanks!

Reply:
This was actually a driving reason.  We want command count files for every trace; they're useful for developer analysis, and I also want to use them for `gfxrecon-convert` testing.  So I definitely want any script that an outside vendor might use to contribute a workload to include this in its basic processing.